### ISSUE TITLE
vSphere ubuntu image-builder is failing at containerd : download containerd

### ISSUE THREAD

**Issue by miryalasaigoutham** [aws/eks-anywhere-build-tooling#2125]
Hi Team,

We are trying to build the Ubuntu-based ova, but it is failing with the below error
```
vsphere-iso.vsphere: TASK [containerd : download containerd] ****************************************
    vsphere-iso.vsphere: fatal: [default]: FAILED! => {"changed": false, "checksum_dest": null, "checksum_src": null, "dest": "/tmp/containerd.tar.gz", "elapsed": 0, "msg": "The checksum format is invalid", "url": "null"}
```

Can you please look into it and let me know if I'm missing anything or if you need any other details?

Below is the command used for building the ova
```
image-builder build --os ubuntu --hypervisor vsphere --release-channel 1-24 --vsphere-config vsphere-connection.json
```

Thanks and Regards
Sai Goutham Miryala

**Comment by miryalasaigoutham** [comment]
```
==> vsphere-iso.vsphere: Executing Ansible: ansible-playbook -e packer_build_name="vsphere" -e packer_*****_type=vsphere-iso -e packer_http_addr=10.198.89.228:8198 --ssh-extra-args '-o IdentitiesOnly=yes' --extra-vars containerd_url=null containerd_sha256=null etcd_http_source=https://distro.eks.amazonaws.com/kubernetes-1-24/releases/15/artifacts/etcd/v3.5.7/etcd-linux-amd64-v3.5.7.tar.gz etcd_version=v3.5.7 etcd_sha256=60c97bd224c6ac1509a2db09df2726936c78e16877585e608029b621ce9d0f34 etcdadm_http_source=https://anywhere-assets.eks.amazonaws.com/releases/bundles/36/artifacts/etcdadm/f089d308442c18f487a52d09fd067ae9ac7cd8f2/etcdadm-linux-amd64.tar.gz etcdadm_version=v0.1.5 pause_image=public.ecr.aws/eks-distro/kubernetes/pause:v1.24.12-eks-1-24-15 containerd_additional_settings= containerd_cri_socket=/var/run/containerd/containerd.sock containerd_version=1.6.18 containerd_wasm_shims_url=https://github.com/deislabs/containerd-wasm-shims/releases/download/v0.3.3/containerd-wasm-shims-v1-linux-x86_64.tar.gz containerd_wasm_shims_version=v0.3.3 containerd_wasm_shims_sha256=da84b1c065a58f95a841d39e143cd7115d43e6faedcce7a8782f2942388260d7 containerd_wasm_shims_runtimes="" crictl_url=https://anywhere-assets.eks.amazonaws.com/releases/bundles/36/artifacts/cri-tools/v1.26.0/cri-tools-linux-amd64.tar.gz crictl_sha256=cf1e9bf60b4a4b719b8180807057f05ba93bdcd5dff49848d90675539eae2111 crictl_source_type=http custom_role_names="" firstboot_custom_roles_pre="" firstboot_custom_roles_post="" node_custom_roles_pre="" node_custom_roles_post="" disable_public_repos=false extra_debs="nfs-common xfsprogs" extra_repos="" extra_rpms="" http_proxy= https_proxy= kubeadm_template=etc/kubeadm.yml kubernetes_cni_http_source=https://distro.eks.amazonaws.com/kubernetes-1-24/releases/15/artifacts/plugins kubernetes_cni_http_checksum=sha256:c8f684b6da0370a9781212b9e68e1ec516cd1bfd7d4716a4ae8e1028c04034d1 kubernetes_http_source=https://distro.eks.amazonaws.com/kubernetes-1-24/releases/15/artifacts/kubernetes kubernetes_container_registry=public.ecr.aws/eks-distro kubernetes_rpm_repo=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64 kubernetes_rpm_gpg_key="https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg" kubernetes_rpm_gpg_check=True kubernetes_deb_repo="https://apt.kubernetes.io/ kubernetes-xenial" kubernetes_deb_gpg_key=https://packages.cloud.google.com/apt/doc/apt-key.gpg kubernetes_cni_deb_version= kubernetes_cni_rpm_version= kubernetes_cni_semver=v1.1.1 kubernetes_cni_source_type=http kubernetes_semver=v1.24.12 kubernetes_source_type=http kubernetes_load_additional_imgs=true kubernetes_deb_version=1.23.15-00 kubernetes_rpm_version=1.23.15-0 no_proxy= pip_conf_file= python_path= redhat_epel_rpm=https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm epel_rpm_gpg_key= reenable_public_repos=true remove_extra_repos=false systemd_prefix=/usr/lib/systemd sysusr_prefix=/usr sysusrlocal_prefix=/usr/local load_additional_components=true additional_registry_images=false additional_registry_images_list= additional_url_images=true additional_url_images_list=https://distro.eks.amazonaws.com/kubernetes-1-23/releases/20/artifacts/kubernetes/v1.23.17/bin/linux/amd64/pause.tar additional_executables=false additional_executables_list= additional_executables_destination_path= build_target=virt amazon_ssm_agent_rpm= --extra-vars guestinfo_datasource_slug=https://raw.githubusercontent.com/vmware/cloud-init-vmware-guestinfo guestinfo_datasource_ref=v1.4.1 guestinfo_datasource_script=https://raw.githubusercontent.com/vmware/cloud-init-vmware-guestinfo/v1.4.1/install.sh --extra-vars  --scp-extra-args  -e ansible_ssh_private_key_file=/tmp/ansible-key4208688612 -i /tmp/packer-provisioner-ansible3635409775 /home/image-*****/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-*****/image-*****/images/capi/ansible/node.yml
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: PLAY [all] *********************************************************************
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [Gathering Facts] *********************************************************
    vsphere-iso.vsphere: ok: [default]
    vsphere-iso.vsphere: [WARNING]: Collection community.general does not support Ansible version 2.10.8
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : Put templated sources.list in place] *****************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : Put templated apt.conf.d/90proxy in place when defined] **********
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : perform a dist-upgrade] ******************************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : install baseline dependencies] ***********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : install extra debs] **********************************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [setup : install pinned debs] *********************************************
    vsphere-iso.vsphere: ok: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Ensure overlay module is present] *********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Ensure br_netfilter module is present] ****************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Persist required kernel modules] **********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Set and persist kernel params] ************************************
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.bridge.bridge-nf-call-iptables', 'val': 1})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.bridge.bridge-nf-call-ip6tables', 'val': 1})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.ipv4.ip_forward', 'val': 1})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.ipv6.conf.all.forwarding', 'val': 1})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.ipv6.conf.all.disable_ipv6', 'val': 0})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'net.ipv4.tcp_congestion_control', 'val': 'bbr'})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'vm.overcommit_memory', 'val': 1})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'kernel.panic', 'val': 10})
    vsphere-iso.vsphere: changed: [default] => (item={'param': 'kernel.panic_on_oops', 'val': 1})
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Ensure auditd is running and comes on at reboot] ******************
    vsphere-iso.vsphere: ok: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : configure auditd rules for containerd] ****************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Ensure reverse packet filtering is set as strict] *****************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Copy udev etcd network tuning rules] ******************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [node : Copy etcd network tuning script] **********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere: included: /home/image-builder/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-builder/image-builder/images/capi/ansible/roles/providers/tasks/vmware.yml for default
    vsphere-iso.vsphere: included: /home/image-builder/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-builder/image-builder/images/capi/ansible/roles/providers/tasks/vmware-ubuntu.yml for default
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Install cloud-init packages] *********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Disable Hyper-V KVP protocol daemon on Ubuntu] ***************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Copy networkd-dispatcher scripts to add DHCP provided NTP servers] ***
    vsphere-iso.vsphere: changed: [default] => (item={'src': 'files/etc/networkd-dispatcher/routable.d/20-chrony.j2', 'dest': '/etc/networkd-dispatcher/routable.d/20-chrony'})
    vsphere-iso.vsphere: changed: [default] => (item={'src': 'files/etc/networkd-dispatcher/off.d/20-chrony.j2', 'dest': '/etc/networkd-dispatcher/off.d/20-chrony'})
    vsphere-iso.vsphere: changed: [default] => (item={'src': 'files/etc/networkd-dispatcher/no-carrier.d/20-chrony.j2', 'dest': '/etc/networkd-dispatcher/no-carrier.d/20-chrony'})
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Create provider vmtools config drop-in file] *****************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Create service to modify cloud-init config] ******************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Copy cloud-init modification script] *************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Enable modify-cloud-init-cfg.service] ************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Creates unit file directory for cloud-final] *****************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Create cloud-final boot order drop in file] ******************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Creates unit file directory for cloud-config] ****************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Create cloud-final boot order drop in file] ******************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Make sure all cloud init services are enabled] ***************
    vsphere-iso.vsphere: ok: [default] => (item=cloud-final)
    vsphere-iso.vsphere: ok: [default] => (item=cloud-config)
    vsphere-iso.vsphere: ok: [default] => (item=cloud-init)
    vsphere-iso.vsphere: ok: [default] => (item=cloud-init-local)
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Create cloud-init config file] *******************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : set cloudinit feature flags] *********************************
    vsphere-iso.vsphere: changed: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [providers : Ensure chrony is running] ************************************
    vsphere-iso.vsphere: ok: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [containerd : Install libseccomp2 package] ********************************
    vsphere-iso.vsphere: ok: [default]
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: TASK [containerd : download containerd] ****************************************
    vsphere-iso.vsphere: fatal: [default]: FAILED! => {"changed": false, "checksum_dest": null, "checksum_src": null, "dest": "/tmp/containerd.tar.gz", "elapsed": 0, "msg": "The checksum format is invalid", "url": "null"}
    vsphere-iso.vsphere:
    vsphere-iso.vsphere: PLAY RECAP *********************************************************************
    vsphere-iso.vsphere: default                    : ok=34   changed=26   unreachable=0    failed=1    skipped=175  rescued=0    ignored=0
    vsphere-iso.vsphere:
==> vsphere-iso.vsphere: Provisioning step had errors: Running the cleanup provisioner, if present...
==> vsphere-iso.vsphere: Clear boot order...
==> vsphere-iso.vsphere: Power off VM...
==> vsphere-iso.vsphere: Deleting Floppy image ...
==> vsphere-iso.vsphere: Destroying VM...
Build 'vsphere-iso.vsphere' errored after 10 minutes 7 seconds: Error executing Ansible: Non-zero exit status: exit status 2

==> Wait completed after 10 minutes 7 seconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso.vsphere: Error executing Ansible: Non-zero exit status: exit status 2

==> Builds finished but no artifacts were created.
make[1]: *** [Makefile:408: build-node-ova-vsphere-ubuntu-2004] Error 1
make[1]: Leaving directory '/home/image-builder/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-builder/image-builder/images/capi'
make: *** [Makefile:283: local-build-ova-ubuntu-2004] Error 2
make: Leaving directory '/home/image-builder/eks-anywhere-build-tooling/projects/kubernetes-sigs/image-builder'
2023/04/26 20:56:08 Error executing image-builder for vsphere hypervisor: failed to run command: exit status 2

```

**Comment by abhay-krishna** [comment]
Thanks for reporting @miryalasaigoutham. We are aware of this issue, this was raised in aws/eks-anywhere#5723 as well. We put in a temp fix at #2134, could you try re-running the build.

**Comment by miryalasaigoutham** [comment]
Thanks @abhay-krishna
